src/annotator/highlighter.js

```diff
@@ -264,6 +274,57 @@ export function removeHighlights(highlights) {
   }
 }
 
+/**
+ * Set whether the given highlight elements should appear "focused".
+ *
+ * A highlight can be displayed in a different ("focused") style to indicate
+ * that it is current in some other context - for example the user has selected
+ * the corresponding annotation in the sidebar.
+ *
+ * @param {HighlightElement[]} highlights
+ * @param {boolean} focused
+ */
+export function setHighlightsFocused(highlights, focused) {
+  highlights.forEach(h =>
+    h.classList.toggle('hypothesis-highlight-focused', focused)
+  );
+}
+
+/**
+ * Set whether highlights under the given root element should be visible.
+ *
+ * @param {HTMLElement} root
+ * @param {boolean} visible
+ */
+export function setHighlightsVisible(root, visible) {
+  const showHighlightsClass = 'hypothesis-highlights-always-on';
+  root.classList.toggle(showHighlightsClass, visible);
+}
+
+/**
+ * Get the highlight elements that contain the given node.
+ *
+ * @param {Node} node
+ * @return {HighlightElement[]}
+ */
+export function getHighlightsContainingNode(node) {
+  let el =
```

Comment:
Is there any chance `node` could be something other than a text node or an element node (does it even matter)? I suppose we're trying to handle for the text node case here...

Reply:
> Is there any chance node could be something other than a text node or an element node (does it even matter)?

In practice `node` will often come from the target of a mouse event, so it could be whatever node types can receive such events. I forget, but I believe that might include embedded SVG or MathML documents.